macos/FluentUI/Button/Button.swift

```diff
@@ -154,6 +154,39 @@ open class Button: NSButton {
 		}
 	}
 
+	/// When set to a non `nil` value ,the Secondary Image will always be placed on the trailing edge of the button in both LTR and
+	/// RTL language settings. It is designed to remain independent of  the `imagePosition` value of the NSButton which deals
+	/// with the main Button `image`.
+	public var secondaryImage: NSImage? {
+			didSet {
+				guard oldValue != secondaryImage else {
+					return
+				}
+
+				guard let cell = cell as? ButtonCell else {
+					return
+				}
+				
+				// Handle replacement/removal of a previously set image
+				if secondaryImageView.isDescendant(of: self) {
```

Comment:
will secondaryImageView ever exist and not be a descendant? Per other comment about making it optional, could then just always remove from superview if it exists?

Reply:
Or actually in the case where it already exists we could just avoid re-adding to the hierarchy and instead just change the image of the NSImageView?